source/server/admin/prometheus_stats.cc

```diff
@@ -11,19 +12,15 @@ namespace Server {
 namespace {
 
 const std::regex& promRegex() { CONSTRUCT_ON_FIRST_USE(std::regex, "[^a-zA-Z0-9_]"); }
+const std::regex& namespaceRegex() { CONSTRUCT_ON_FIRST_USE(std::regex, "^[a-zA-Z0-9]*$"); }
 
 /**
  * Take a string and sanitize it according to Prometheus conventions.
  */
 std::string sanitizeName(const std::string& name) {
   // The name must match the regex [a-zA-Z_][a-zA-Z0-9_]* as required by
   // prometheus. Refer to https://prometheus.io/docs/concepts/data_model/.
```

Comment:
Can you add onto this comment to say that the initial `[a-zA-Z_]` constraint is always satisfied by the namespace/prefix? (I'm assuming that's why you deleted the number-detection chunk below.)